### PR TITLE
[Reviewer: Krista] Add ifcs plugins

### PR DESCRIPTION
**Pull request by eleanor-merry**
Adds the plugins and scripts to share the ifc configuration around

### CODE REVIEW

File: clearwater_config_manager/default_ifcs_config_plugin.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# Project Clearwater - IMS in the Cloud
+# Copyright (C) 2017  Metaswitch Networks Ltd
```
Comment (Krista-Rogers) [inline clearwater_config_manager/default_ifcs_config_plugin.py]: Switch out for new copyright statement
(and same for all other new files in this pull request)

Comment (Krista-Rogers) [inline clearwater_config_manager/default_ifcs_config_plugin.py]: Missing "@file file_name" at the top of your new files.

File: clearwater_config_manager/default_ifcs_config_plugin.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                else:
+                    return FileStatus.OUT_OF_SYNC
+        except IOError:  # pragma: no cover
+            return FileStatus.MISSING
+
+    def on_config_changed(self, value, alarm):
+        _log.info("Updating Default IFCs configuration file")
+
+        if self.status(value) != FileStatus.UP_TO_DATE:
+            safely_write(_file, value)
```
Comment (Krista-Rogers) [inline clearwater_config_manager/default_ifcs_config_plugin.py]: Might be a stupid question but.... could we get stuck in a loop here?
(if the status is OUT_OF_SYNC this shouldn't happen, but if the status is MISSING I'm concerned we might loop)
(I also have the same worry for the other config plugin)

File: clearwater_config_manager/scripts/config_validation/shared_ifcs_schema.xsd [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+  <xs:complexType name="tIncludeRegisterResponse">
+    <xs:sequence>
+      <xs:element name="Extension" type="tExtension" minOccurs="0"/>
+      <xs:any namespace="##other" processContents="lax" minOccurs="0" maxOccurs="unbounded"/>
+    </xs:sequence>
+  </xs:complexType>
+  <xs:element name="SharedIFCsSets">
+  <xs:complexType>
+    <xs:sequence>
+        <xs:element name="SharedIFCsSet" maxOccurs="unbounded">
```
Comment (Krista-Rogers) [inline clearwater_config_manager/scripts/config_validation/shared_ifcs_schema.xsd]: Does this need a minOccurs of 0?

File: clearwater_config_manager/scripts/remove_default_ifcs_config [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+local_site_name=site1
+etcd_key=clearwater
+. /etc/clearwater/config
+. /usr/share/clearwater/utils/check-root-permissions 1
+
+# Check we can contact `etcd`
+if ! nc -z ${management_local_ip:-$local_ip} 4000
+then
+  echo "The Clearwater Configuration store (etcd) is not running"
+  echo "Please start it before removing configuration"
```
Comment (Krista-Rogers) [inline clearwater_config_manager/scripts/remove_default_ifcs_config]: Should we give a pointer to how they start it? (Or is this something they'll know?)
(same comment in other files)

**Review by Krista-Rogers — APPROVED** [review]
Looks great!
I've only left a few comments :)
(Also it matches the spec perfectly!)

File: clearwater_config_manager/scripts/config_validation/shared_ifcs_schema.xsd (hunk 3 above)
Comment (eleanor-merry) [reply clearwater_config_manager/scripts/config_validation/shared_ifcs_schema.xsd]: Good spot, added.

**Pull request merged**